### PR TITLE
Rework hitbox

### PR DESCRIPTION
**Pull request by Anchousmsu**
Issue #50

### CODE REVIEW

File: src/Model/CommonTypes.hs [hunk 1]
```diff
@@ -68,15 +68,11 @@ computeScale :: Rectangle
...
-  { hitboxPosition :: Position -- ^ Object position
-  , hitboxCollisionBoxes :: [Rectangle]
-  } deriving Show
-
 -- | Contains information about a movable game entity.
 data Object = Object
   { objectName :: String -- ^ Unique object identifier.
-  , objectHitbox :: Hitbox -- ^ Object hitbox.
+  , objectPosition :: Position -- ^ Object position
+  , objectCollision :: [Rectangle]
```
Comment (alois9866) [inline src/Model/CommonTypes.hs]: Rename to objectCollisionBoxes.

File: src/Model/CommonTypes.hs [hunk 2]
```diff
@@ -68,15 +68,11 @@ computeScale :: Rectangle
...
-  { hitboxPosition :: Position -- ^ Object position
-  , hitboxCollisionBoxes :: [Rectangle]
-  } deriving Show
-
 -- | Contains information about a movable game entity.
 data Object = Object
   { objectName :: String -- ^ Unique object identifier.
-  , objectHitbox :: Hitbox -- ^ Object hitbox.
+  , objectPosition :: Position -- ^ Object position
+  , objectCollisionBoxes :: [Rectangle]
```
Comment (dm-gh) [inline src/Model/CommonTypes.hs]: I like "objectHitbox" name more than "objectCollisionBoxes". It's shorter and clearer

Comment (Anchousmsu) [reply src/Model/CommonTypes.hs]: but "objectHitbox" less understandable. so I insist on leaving "objectCollisionBoxes"

Comment (dm-gh) [reply src/Model/CommonTypes.hs]: Ok

**Pull request merged**